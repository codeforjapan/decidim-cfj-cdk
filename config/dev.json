{
  "rds": {
    "snapshot": true,
    "snapshotIdentifier": "decidim-master-2025-03-12",
    "instanceType": "t3.micro",
    "deletionProtection": false,
    "allocatedStorage": 20,
    "maxAllocatedStorage": 40,
    "enablePerformanceInsights": false,
    "multiAz": false
  },

  "aws": {
    "accountId": "887442827229",
    "region": "ap-northeast-1"
  },

  "cacheNodeType": "cache.t2.micro",
  "engineVersion": "6.x",
  "numCacheNodes": 1,
  "automaticFailoverEnabled": false,

  "ecs": {
    "smtpDomain": "diycities.jp",
    "repository": "decidim-cfj",
    "certificates": [
      "arn:aws:acm:ap-northeast-1:887442827229:certificate/faa7a120-3843-4561-93ec-c3d313356ad5",
      "arn:aws:acm:ap-northeast-1:887442827229:certificate/b02c7415-d6f8-4794-994e-fe5689ca7b74"
    ],
    "fargateSpotCapacityProvider": {
      "base": 1,
      "weight":2
    },
    "fargateCapacityProvider": {
      "weight":1
    }
  },

  "domain": "diycities.jp",
  "cloudfrontCertificate": "arn:aws:acm:us-east-1:887442827229:certificate/8aec1d57-e068-47f3-a180-e0c1fbb782de"
}
