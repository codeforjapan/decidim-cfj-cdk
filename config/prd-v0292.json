{
  "rds": {
    "snapshot": true,
    "snapshotIdentifier": "decidim-master-2025-07-14",
    "instanceType": "t3.medium",
    "deletionProtection": true,
    "allocatedStorage": 20,
    "maxAllocatedStorage": 40,
    "enablePerformanceInsights": true,
    "multiAz": true
  },

  "aws": {
    "accountId": "887442827229",
    "region": "ap-northeast-1"
  },

  "cacheNodeType": "cache.t3.medium",
  "engineVersion": "6.x",
  "numCacheNodes": 3,
  "automaticFailoverEnabled": true,

  "ecs": {
    "smtpDomain": "diycities.jp",
    "repository": "decidim-cfj",
    "certificates": [
      "arn:aws:acm:ap-northeast-1:887442827229:certificate/faa7a120-3843-4561-93ec-c3d313356ad5",
      "arn:aws:acm:ap-northeast-1:887442827229:certificate/b02c7415-d6f8-4794-994e-fe5689ca7b74"
    ],
    "fargateSpotCapacityProvider": {
      "weight": 1
    },
    "fargateCapacityProvider": {
      "base": 1,
      "weight": 2
    }
  },

  "domain": "diycities.jp",
  "cloudfrontCertificate": "arn:aws:acm:us-east-1:887442827229:certificate/8aec1d57-e068-47f3-a180-e0c1fbb782de"
}
