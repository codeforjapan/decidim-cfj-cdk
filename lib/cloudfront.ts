import {
  aws_certificatemanager,
  aws_cloudfront as cloudfront,
  aws_cloudfront_origins,
  aws_wafv2,
  Stack
} from "aws-cdk-lib";
import { Construct } from "constructs";
import { BaseStackProps } from "./props";
import { AllowedMethods, CachePolicy, OriginRequestPolicy, ViewerProtocolPolicy } from "aws-cdk-lib/aws-cloudfront";

export interface CloudFrontProps extends BaseStackProps {
  certificateArn: string
  domain: string
}

export class CloudFrontStack extends Stack {
  constructor(scope: Construct, id: string, props: CloudFrontProps) {
    super(scope, id, props);
    const endpoint = `${ props.stage }-${ props.serviceName }-alb-origin.${ props.domain }`;
    const origin = new aws_cloudfront_origins.HttpOrigin(endpoint)
    origin.bind(this, { originId: "defaultEndPoint" })

    const rules: any[] = [
      {
        name: `${ props.stage }-${ props.serviceName }-AWSManagedRulesCommonRuleSet`,
        priority: 1,
        statement: {
          managedRuleGroupStatement: {
            vendorName: "AWS",
            name: "AWSManagedRulesCommonRuleSet",
            excludedRules: [
              { name: 'CrossSiteScripting_BODY' },
              { name: 'SizeRestrictions_BODY' },
              { name: 'GenericRFI_BODY' }
            ]
          },
        },
        overrideAction: { none: {} },
        visibilityConfig: {
          cloudWatchMetricsEnabled: true,
          sampledRequestsEnabled: true,
          metricName: `${ props.stage }-${ props.serviceName }-AWSManagedRulesCommonRuleSet`,
        },
      },
      {
        name: `${ props.stage }-${ props.serviceName }-AWSManagedRulesKnownBadInputsRuleSet`,
        priority: 2,
        statement: {
          managedRuleGroupStatement: {
            vendorName: "AWS",
            name: "AWSManagedRulesKnownBadInputsRuleSet",
          },
        },
        overrideAction: { none: {} },
        visibilityConfig: {
          cloudWatchMetricsEnabled: true,
          sampledRequestsEnabled: true,
          metricName: `${ props.stage }-${ props.serviceName }-AWSManagedRulesKnownBadInputsRuleSet`,
        },
      },
      {
        name: `${ props.stage }-${ props.serviceName }-AWSManagedRulesAmazonIpReputationList`,
        priority: 3,
        statement: {
          managedRuleGroupStatement: {
            vendorName: "AWS",
            name: "AWSManagedRulesAmazonIpReputationList",
          },
        },
        overrideAction: { none: {} },
        visibilityConfig: {
          cloudWatchMetricsEnabled: true,
          sampledRequestsEnabled: true,
          metricName: `${ props.stage }-${ props.serviceName }-AWSManagedRulesAmazonIpReputationList`,
        },
      },
      {
        name: `${ props.stage }-${ props.serviceName }-AWSManagedRulesLinuxRuleSet`,
        priority: 4,
        statement: {
          managedRuleGroupStatement: {
            vendorName: "AWS",
            name: "AWSManagedRulesLinuxRuleSet",
          },
        },
        overrideAction: { none: {} },
        visibilityConfig: {
          cloudWatchMetricsEnabled: true,
          sampledRequestsEnabled: true,
          metricName: `${ props.stage }-${ props.serviceName }-AWSManagedRulesLinuxRuleSet`,
        },
      },
      {
        name: `${ props.stage }-${ props.serviceName }-AWSManagedRulesSQLiRuleSet`,
        priority: 5,
        statement: {
          managedRuleGroupStatement: {
            vendorName: "AWS",
            name: "AWSManagedRulesSQLiRuleSet",
            excludedRules: [
              { name: 'SQLi_BODY' }
            ]
          },
        },
        overrideAction: { none: {} },
        visibilityConfig: {
          cloudWatchMetricsEnabled: true,
          sampledRequestsEnabled: true,
          metricName: `${ props.stage }-${ props.serviceName }-AWSManagedRulesSQLiRuleSet`,
        },
      },
      {
        name: `${ props.stage }-${ props.serviceName }-AllowSystemLogin`,
        priority: 6,
        statement: {
          andStatement: {
            statements: [{
              byteMatchStatement: {
                searchString: `${ props.stage }-${ props.serviceName }-alb-origin`,
                fieldToMatch: {
                  singleHeader: {
                    name: 'host'
                  }
                },
                textTransformations: [
                  {
                    priority: 0,
                    type: 'LOWERCASE'
                  }
                ],
                positionalConstraint: "EXACTLY"
              }
            },
              {
                byteMatchStatement: {
                  searchString: "system/admins/sign_in",
                  fieldToMatch: {
                    uriPath: {}
                  },
                  textTransformations: [
                    {
                      priority: 0,
                      type: 'LOWERCASE'
                    }
                  ],
                  positionalConstraint: "CONTAINS"
                }
              }
            ]
          }
        },
        action: {
          allow: {}
        },
        visibilityConfig: {
          cloudWatchMetricsEnabled: true,
          sampledRequestsEnabled: true,
          metricName: `${ props.stage }-${ props.serviceName }-AllowSystemLogin`,
        },
      }
    ]

    if (props.stage === 'prd-v0283') {
      rules.push({
        name: 'production-AllowSystemLogin',
        priority: 7,
        statement: {
          andStatement: {
            statements: [{
              byteMatchStatement: {
                searchString: 'production.diycities.jp',
                fieldToMatch: {
                  singleHeader: {
                    name: 'host'
                  }
                },
                textTransformations: [
                  {
                    priority: 0,
                    type: 'LOWERCASE'
                  }
                ],
                positionalConstraint: "EXACTLY"
              }
            },
              {
                byteMatchStatement: {
                  searchString: "system/admins/sign_in",
                  fieldToMatch: {
                    uriPath: {}
                  },
                  textTransformations: [
                    {
                      priority: 0,
                      type: 'LOWERCASE'
                    }
                  ],
                  positionalConstraint: "CONTAINS"
                }
              }
            ]
          }
        },
        action: {
          allow: {}
        },
        visibilityConfig: {
          cloudWatchMetricsEnabled: true,
          sampledRequestsEnabled: true,
          metricName: `production-AllowSystemLogin`,
        },
      })
    }

    rules.push({
      name: `${ props.stage }-${ props.serviceName }-SystemLoginBlock`,
      priority: 8,
      statement: {
        byteMatchStatement: {
          searchString: 'system/admins/sign_in',
          fieldToMatch: {
            uriPath: {}
          },
          textTransformations: [
            {
              priority: 0,
              type: "LOWERCASE"
            }
          ],
          positionalConstraint: "CONTAINS"
        }
      },
      action: {
        block: {
          customResponse: {
            responseCode: 403,
            customResponseBodyKey: "disable-action"
          }
        }
      },
      visibilityConfig: {
        cloudWatchMetricsEnabled: true,
        sampledRequestsEnabled: true,
        metricName: `${ props.stage }-${ props.serviceName }-AllowSystemLogin`,
      },
    })

    const waf: aws_wafv2.CfnWebACL = new aws_wafv2.CfnWebACL(this, "CfnWebACL", {
        name: `${ props.stage }-${ props.serviceName }-webAcl`,
        defaultAction: { allow: {} },
        scope: "CLOUDFRONT",
        visibilityConfig: {
          cloudWatchMetricsEnabled: true,
          sampledRequestsEnabled: true,
          metricName: `${ props.stage }-${ props.serviceName }-webAcl-metrics`
        },
        description: `Web ACL for ${ props.stage }-${ props.serviceName }-cloudfront`,
        rules: rules,
        customResponseBodies: {
          'disable-action': {
            content: '<div>error: access denied</div>',
            contentType: 'TEXT_HTML'
          }
        }
      }
    )

    if (props.stage === 'prd-v0283') {
      new cloudfront.Distribution(this, 'Distribution', {
        priceClass: cloudfront.PriceClass.PRICE_CLASS_ALL,
        defaultBehavior: {
          origin: origin,
          allowedMethods: AllowedMethods.ALLOW_ALL,
          viewerProtocolPolicy: ViewerProtocolPolicy.REDIRECT_TO_HTTPS,
          cachePolicy: CachePolicy.CACHING_DISABLED,
          originRequestPolicy: OriginRequestPolicy.ALL_VIEWER
        },
        comment: `${ props.stage }-${ props.serviceName }-cloudfront`,
        domainNames: [endpoint, `*.${ props.domain }`],
        certificate: aws_certificatemanager.Certificate.fromCertificateArn(this, 'cloudFrontCertificate', props.certificateArn),
        webAclId: waf.attrArn
      }).addBehavior('decidim-packs/*', origin, {
        allowedMethods: AllowedMethods.ALLOW_GET_HEAD_OPTIONS,
        viewerProtocolPolicy: ViewerProtocolPolicy.REDIRECT_TO_HTTPS,
        cachePolicy: CachePolicy.CACHING_OPTIMIZED
      })
    } else {
      new cloudfront.Distribution(this, 'Distribution', {
        priceClass: cloudfront.PriceClass.PRICE_CLASS_ALL,
        defaultBehavior: {
          origin: origin,
          allowedMethods: AllowedMethods.ALLOW_ALL,
          viewerProtocolPolicy: ViewerProtocolPolicy.REDIRECT_TO_HTTPS,
          cachePolicy: CachePolicy.CACHING_DISABLED,
          originRequestPolicy: OriginRequestPolicy.ALL_VIEWER
        },
        comment: `${ props.stage }-${ props.serviceName }-cloudfront`,
        domainNames: [endpoint],
        certificate: aws_certificatemanager.Certificate.fromCertificateArn(this, 'cloudFrontCertificate', props.certificateArn),
        webAclId: waf.attrArn
      }).addBehavior('decidim-packs/*', origin, {
        allowedMethods: AllowedMethods.ALLOW_GET_HEAD_OPTIONS,
        viewerProtocolPolicy: ViewerProtocolPolicy.REDIRECT_TO_HTTPS,
        cachePolicy: CachePolicy.CACHING_OPTIMIZED
      })
    }
  }
}
