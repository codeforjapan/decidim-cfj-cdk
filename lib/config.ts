import { InstanceType } from "aws-cdk-lib/aws-ec2";

export interface SubnetConfig {
  subnetId: string;
  availabilityZone: string;
  routeTableId: string;
}

export interface VpcConfig {
  vpcId: string;
  cidrBlock: string;
  availabilityZones: string[];
  publicSubnets: SubnetConfig[];
  privateSubnets: SubnetConfig[];
}

export interface RdsConfig {
  snapshot: boolean
  snapshotIdentifier: string
  instanceType: InstanceType
  deletionProtection: boolean
  allocatedStorage: number
  maxAllocatedStorage: number
  multiAz: boolean
  enablePerformanceInsights: boolean
}

export interface EcsConfig {
  smtpDomain: string,
  repository: string
  certificates: string[]
  fargateCapacityProvider: capacityProviderStrategy
  fargateSpotCapacityProvider: capacityProviderStrategy
}

export interface capacityProviderStrategy {
  base?: number;
  weight: number;
}

export interface Config {
  stage: string

  vpc?: VpcConfig;
  aws: {
    accountId: string;
    region: string;
  };

  s3Bucket: string

  // rds
  rds: RdsConfig

  // elastiCache
  cacheNodeType: string
  engineVersion: string
  numCacheNodes: number
  automaticFailoverEnabled: boolean

  ecs: EcsConfig

  // service
  domain: string,
  cloudfrontCertificate: string
}

export function getConfig (stage: string): Config {
  return require(`../config/${ stage }.json`)
}
