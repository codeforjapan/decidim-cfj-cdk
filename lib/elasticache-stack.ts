import { aws_elasticache as elasticache, Stack } from "aws-cdk-lib";

import { Construct } from "constructs";
import { BaseStackProps } from "./props";
import { CfnReplicationGroup, CfnReplicationGroupProps } from "aws-cdk-lib/aws-elasticache";

export interface ElastiCacheStackProps extends BaseStackProps {
  cacheNodeType: string
  engineVersion: string
  numCacheNodes: number
  automaticFailoverEnabled: boolean
  securityGroup: string
  ecSubnetGroup: elasticache.CfnSubnetGroup
}

export class ElasticacheStack extends Stack {
  public readonly redis: CfnReplicationGroup;

  constructor(scope: Construct, id: string, props: ElastiCacheStackProps) {
    super(scope, id, props);

    const elastiCacheProps: CfnReplicationGroupProps = {
      replicationGroupDescription: `${ props.stage }-${ props.serviceName }-cache`,
      engine: 'redis',
      replicationGroupId: `${ props.stage }-${ props.serviceName }-cache`,
      engineVersion: props.engineVersion,
      cacheNodeType: props.cacheNodeType,
      numCacheClusters: props.numCacheNodes,
      automaticFailoverEnabled: props.automaticFailoverEnabled,
      securityGroupIds: [props.securityGroup],
      cacheSubnetGroupName: props.ecSubnetGroup.cacheSubnetGroupName,
    }

    if (props.stage === 'prd-v0283') {
      this.redis = new elasticache.CfnReplicationGroup(this, 'prdElasticache', {
        ...elastiCacheProps,
        ...{
          multiAzEnabled: true,
        }
      })
    } else {
      this.redis = new elasticache.CfnReplicationGroup(this, 'elasticache', elastiCacheProps)
    }
  }
}
