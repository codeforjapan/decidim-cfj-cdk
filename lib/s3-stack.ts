import { aws_cloudfront as cloudfront, aws_iam as iam, aws_s3, RemovalPolicy, Stack } from "aws-cdk-lib";
import { Construct } from "constructs";
import { BaseStackProps } from "./props";
import { HttpMethods } from "aws-cdk-lib/aws-s3";

export class S3Stack extends Stack {
  public readonly bucket: aws_s3.Bucket

  constructor(scope: Construct, id: string, props: BaseStackProps) {
    super(scope, id, props);

    const bucket = new aws_s3.Bucket(this, 'createBucket', {
      bucketName: `${ props.stage }-${ props.serviceName }-bucket`,
      versioned: props.stage === 'prd-v0292',
      removalPolicy: RemovalPolicy.DESTROY,
      autoDeleteObjects: true,
      blockPublicAccess: aws_s3.BlockPublicAccess.BLOCK_ALL,
      cors: [
        {
          allowedHeaders: ['*'],
          allowedMethods: [HttpMethods.PUT, HttpMethods.GET, HttpMethods.POST ],
          allowedOrigins: ['*'],
          exposedHeaders: [
            'Origin',
            'Content-Type',
            'Content-MD5',
            'Content-Disposition'
          ],
          maxAge: 3600
        }
      ]
    });

    this.bucket = bucket

    const oai = new cloudfront.OriginAccessIdentity(
      this,
      `${ props.stage }-${ props.serviceName }-OriginAccessIdentity`,
      {
        comment: `${ props.stage }-${ props.serviceName }-OriginAccessIdentity`
      }
    )

    bucket.addToResourcePolicy(
      new iam.PolicyStatement({
        actions: ["s3:GetObject"],
        effect: iam.Effect.ALLOW,
        principals: [
          new iam.CanonicalUserPrincipal(
            oai.cloudFrontOriginAccessIdentityS3CanonicalUserId
          )
        ],
        resources: [`${ bucket.bucketArn }/*`],
      })
    )
  }
}
