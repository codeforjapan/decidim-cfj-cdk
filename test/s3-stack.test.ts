import * as cdk from 'aws-cdk-lib';
import { Template } from 'aws-cdk-lib/assertions';

import { Config, getConfig } from "../lib/config";
import { S3Stack } from "../lib/s3-stack";

test('S3Stack Created', () => {
    const app = new cdk.App();

    const stage = 'staging';
    const config: Config = getConfig(stage)
    const serviceName = `decidim`;

    const env = {
        account: config.aws.accountId,
        region: config.aws.region
    }

    const s3stack = new S3Stack(app, `${ stage }${ serviceName }S3Stack`, {
        stage,
        env,
        serviceName,
        bucketName: config.s3Bucket,
    })

    const template = Template.fromStack(s3stack);

    // Assert the template matches the snapshot.
    expect(template.toJSON()).toMatchSnapshot();
});

