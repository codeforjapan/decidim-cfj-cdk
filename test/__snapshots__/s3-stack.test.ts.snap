// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`S3Stack Created 1`] = `
{
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "CustomS3AutoDeleteObjectsCustomResourceProviderHandler9D90184F": {
      "DependsOn": [
        "CustomS3AutoDeleteObjectsCustomResourceProviderRole3B1BD092",
      ],
      "Properties": {
        "Code": {
          "S3Bucket": "cdk-hnb659fds-assets-887442827229-ap-northeast-1",
          "S3Key": "HASH-REPLACED.zip",
        },
        "Description": {
          "Fn::Join": [
            "",
            [
              "Lambda function for auto-deleting objects in ",
              {
                "Ref": "createBucketC28D3C56",
              },
              " S3 bucket.",
            ],
          ],
        },
        "Handler": "index.handler",
        "MemorySize": 128,
        "Role": {
          "Fn::GetAtt": [
            "CustomS3AutoDeleteObjectsCustomResourceProviderRole3B1BD092",
            "Arn",
          ],
        },
        "Runtime": "nodejs20.x",
        "Timeout": 900,
      },
      "Type": "AWS::Lambda::Function",
    },
    "CustomS3AutoDeleteObjectsCustomResourceProviderRole3B1BD092": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Sub": "arn:\${AWS::Partition}:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "createBucketAutoDeleteObjectsCustomResourceDB0134B0": {
      "DeletionPolicy": "Delete",
      "DependsOn": [
        "createBucketPolicyB2658FD7",
      ],
      "Properties": {
        "BucketName": {
          "Ref": "createBucketC28D3C56",
        },
        "ServiceToken": {
          "Fn::GetAtt": [
            "CustomS3AutoDeleteObjectsCustomResourceProviderHandler9D90184F",
            "Arn",
          ],
        },
      },
      "Type": "Custom::S3AutoDeleteObjects",
      "UpdateReplacePolicy": "Delete",
    },
    "createBucketC28D3C56": {
      "DeletionPolicy": "Delete",
      "Properties": {
        "BucketName": "staging-decidim-bucket",
        "CorsConfiguration": {
          "CorsRules": [
            {
              "AllowedHeaders": [
                "*",
              ],
              "AllowedMethods": [
                "PUT",
                "GET",
                "POST",
              ],
              "AllowedOrigins": [
                "*",
              ],
              "ExposedHeaders": [
                "Origin",
                "Content-Type",
                "Content-MD5",
                "Content-Disposition",
              ],
              "MaxAge": 3600,
            },
          ],
        },
        "PublicAccessBlockConfiguration": {
          "BlockPublicAcls": true,
          "BlockPublicPolicy": true,
          "IgnorePublicAcls": true,
          "RestrictPublicBuckets": true,
        },
        "Tags": [
          {
            "Key": "aws-cdk:auto-delete-objects",
            "Value": "true",
          },
        ],
      },
      "Type": "AWS::S3::Bucket",
      "UpdateReplacePolicy": "Delete",
    },
    "createBucketPolicyB2658FD7": {
      "Properties": {
        "Bucket": {
          "Ref": "createBucketC28D3C56",
        },
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "s3:PutBucketPolicy",
                "s3:GetBucket*",
                "s3:List*",
                "s3:DeleteObject*",
              ],
              "Effect": "Allow",
              "Principal": {
                "AWS": {
                  "Fn::GetAtt": [
                    "CustomS3AutoDeleteObjectsCustomResourceProviderRole3B1BD092",
                    "Arn",
                  ],
                },
              },
              "Resource": [
                {
                  "Fn::GetAtt": [
                    "createBucketC28D3C56",
                    "Arn",
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      {
                        "Fn::GetAtt": [
                          "createBucketC28D3C56",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
            {
              "Action": "s3:GetObject",
              "Effect": "Allow",
              "Principal": {
                "CanonicalUser": {
                  "Fn::GetAtt": [
                    "stagingdecidimOriginAccessIdentity601BDE4E",
                    "S3CanonicalUserId",
                  ],
                },
              },
              "Resource": {
                "Fn::Join": [
                  "",
                  [
                    {
                      "Fn::GetAtt": [
                        "createBucketC28D3C56",
                        "Arn",
                      ],
                    },
                    "/*",
                  ],
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::S3::BucketPolicy",
    },
    "stagingdecidimOriginAccessIdentity601BDE4E": {
      "Properties": {
        "CloudFrontOriginAccessIdentityConfig": {
          "Comment": "staging-decidim-OriginAccessIdentity",
        },
      },
      "Type": "AWS::CloudFront::CloudFrontOriginAccessIdentity",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
